### ISSUE TITLE
IOS 17.3 IPhone 12 Pictures APP can not choose cups printer

### ISSUE THREAD

**Issue by tianmax** [OpenPrinting/cups#884]
**Describe the bug**
IOS 17.3 IPhone 12 Pictures APP can not choose cups printer.

**To Reproduce**
1. Goto iPhone Pictures APP
2. Choose Print, Choose CUPS printer
3. But can not choose this printer

**System Information:**
 - OS and its version: iOS 17.3
 - Application iPhone Apple Pictures APP
 - CUPS version 2.4.7

**Comment by michaelrsweet** [comment]
Do you mean the "Photos" app?

What printer are you trying to print to? What driver are you using?

**Comment by tianmax** [comment]
Yes, HP DJ 2132, HP Laserjet 1020
I used Wireshark to capture packets and found that there were only MDNS requests and responses, without connecting to the TCP 631 interface. It may be a problem with some iPhone models.
Thank you for your reply.

**Comment by gmuth** [comment]
This looks more like a config/setup/network/bonjour-printer-discovery issue to me. You can check our Bonjour setup like this: 

`dns-sd -Z _ipp._tcp`

You should see PTR, SRV and TXT records for your printer.

**Comment by tianmax** [comment]
[photos_mdns.pcapng.zip](https://github.com/OpenPrinting/cups/files/14321449/photos_mdns.pcapng.zip)
Here is pcap files of wireshark, Thanks for help.

**Comment by tianmax** [comment]
[photos_mdns.pcapng.zip](https://github.com/OpenPrinting/cups/files/14321449/photos_mdns.pcapng.zip)
Here is pcap files of wireshark, Thanks for help.

**Comment by michaelrsweet** [comment]
Well, the packet trace you attached shows the client querying and getting the address and TXT records for each of the shared printers, so that part (at least) is working.

It would be useful to see a trace showing the port 631 traffic, as well as a debug log from the CUPS server when you pick one of the printers.

**Comment by tianmax** [comment]
I used Wireshark to capture packets, but after communicating with MDNS, I did not detect any communication with TCP port 631. What fields in the photo app may not have been met in the MDNS response? Or what kind of bug is in the photo app? Other apps are working normally, such as the browser, memo, and so on。
Thank you for your reply.

**Comment by michaelrsweet** [comment]
You should see activity on port 631 once you select one of the CUPS printers, no matter the application. Just browsing the available printers won't trigger an IPP request.

**Comment by tianmax** [comment]
Yes, I know, I have chosen a CUPS printer， But there is no interaction with port 631 in wireshak or cups's log, and this printer cannot be selected on the Photos APP GUI.
Thank you for your reply.
